lib/auth.js

```diff
@@ -48,6 +55,66 @@ Auth.prototype.bearer = function (bearer, sendImmediately) {
   }
 }
 
+
+Auth.prototype.negotiate = function(request, authHeader, sendImmediately) {
+  var self = this
+
+  if (kerberos === undefined) {
+    // Nothing to do if we don't have Kerberos
+    // or a current credential cache
+    return
+  }
+
+  self.hasAuth = true
+
+  if (!sendImmediately) {
```

Comment:
That probably makes sense. I guess `sendImmediately` is set to true by default for basic auth because you are supplying your `user` name and `pass` so it kind of makes sense to send them by default.

> Not all servers use Kerberos, but we want the ability to authenticate to the ones that do.
> By keeping the default to false, we can work with what each server requires.

I'm still not aware what can go so horribly wrong by having `sendImmediately` set to true by default, but anyway, I just added a test case for that https://github.com/simov/request/blob/feature-kerberos-auth/tests/test-kerberos.js#L55-L62